.github/workflows/set-metadata.yml

```diff
@@ -0,0 +1,354 @@
+            details=$(echo "$details" | jq '."howso-validator-py".build_title="${{ needs.get-howso-validator-py-details.outputs.build-title }}"')
+            details=$(echo "$details" | jq '."howso-validator-py".head_sha="${{ needs.get-howso-validator-py-details.outputs.head-sha }}"')
+            details=$(echo "$details" | jq '."howso-validator-py".url="${{ needs.get-howso-validator-py-details.outputs.url }}"')
+          fi
+          if ! [[ -n "${{ needs.get-howso-validator-enterprise-py-details.outputs.run-id }}" ]]; then
+            # Append run-id, run-type, build-date, build-title, head-sha, url to metadata JSON
+            details=$(echo "$details" | jq '. + {"howso-validator-enterprise-py": "{}"}')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".run_id="${{ needs.get-howso-validator-enterprise-py-details.outputs.run-id }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".run_type="${{ needs.get-howso-validator-enterprise-py-details.outputs.run-type }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".build_date="${{ needs.get-howso-validator-enterprise-py-details.outputs.build-date }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".build_title="${{ needs.get-howso-validator-enterprise-py-details.outputs.build-title }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".head_sha="${{ needs.get-howso-validator-enterprise-py-details.outputs.head-sha }}"')
+            details=$(echo "$details" | jq '."howso-validator-enterprise-py".url="${{ needs.get-howso-validator-enterprise-py-details.outputs.url }}"')
+          fi
+          if ! [[ -n "${{ needs.get-howso-synthesizer-py-details.outputs.run-id }}" ]]; then
+            # Append run-id, run-type, build-date, build-title, head-sha, url to metadata JSON
+            details=$(echo "$details" | jq '. + {"howso-engine-py": "{}"}')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".run_id="${{ needs.get-howso-synthesizer-py-details.outputs.run-id }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".run_type="${{ needs.get-howso-synthesizer-py-details.outputs.run-type }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".build_date="${{ needs.get-howso-synthesizer-py-details.outputs.build-date }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".build_title="${{ needs.get-howso-synthesizer-py-details.outputs.build-title }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".head_sha="${{ needs.get-howso-synthesizer-py-details.outputs.head-sha }}"')
+            details=$(echo "$details" | jq '."howso-synthesizer-py".url="${{ needs.get-howso-synthesizer-py-details.outputs.url }}"')
+          fi
```

Comment:
Same as others, looks like a good place for a loop.

Reply:
Same issue with accessing workflow inputs dynamically I think, but will update this as well if you know a way